lib/perf_hooks.js

```diff
@@ -0,0 +1,311 @@
+    return ret;
+  }
+
+  getEntriesByName(name, type) {
+    const ret =
+      type === undefined ?
+        getEntries(`${name}`) : getEntries(`${name}`, `${type}`);
+    if (name === 'frame' && (type === undefined || type === 'frame'))
+      ret.unshift(nodeFrame);
+    if (name === 'node' && (type === undefined || type === 'node'))
+      ret.unshift(nodeTiming);
+    return ret;
+  }
+
+  getEntriesByType(type) {
+    const ret = getEntries(undefined, `${type}`);
+    if (type === 'frame')
+      ret.unshift(nodeFrame);
+    if (type === 'node')
+      ret.unshift(nodeTiming);
+    return ret;
+  }
+
+  mark(name) {
+    name = `${name}`;
+    if (marks.has(name)) {
+      resetMark(name);
+    } else {
+      marks.set(name, new PerformanceMark(name));
+    }
+  }
+
+  clearMarks(name) {
+    if (name !== undefined) {
+      // clear individual mark
+      name = `${name}`;
+      const entry = marks.get(name);
+      if (entry === undefined)
+        return;
+      clearEntry(entry);
+      marks.delete(name);
+    } else {
+      // clear all marks
+      marks.forEach(clearEntry);
+      marks.clear();
+    }
+  }
+
+  measure(name, startMark, endMark) {
+    name = `${name}`;
+    endMark = `${endMark}`;
+    startMark = startMark !== undefined ? `${startMark}` : '';
+    if (!marks.has(endMark) && nodeTimingKeys.indexOf(endMark) < 0) {
+      const errors = lazyErrors();
+      throw new errors.Error('ERR_INVALID_PERFORMANCE_MARK', endMark);
+    }
+    let m = measures.get(name);
+    if (m === undefined) {
+      m = [];
+      measures.set(name, m);
+    }
+    m.push(new PerformanceMeasure(name, startMark, endMark));
+  }
+
+  clearMeasures(name) {
+    if (name !== undefined) {
+      name = `${name}`;
+      const list = measures.get(name);
+      if (list !== undefined) {
+        for (var n = 0; n < list.length; n++)
+          clearEntry(list[n]);
+      }
+      measures.delete(name);
+    } else {
+      measures.forEach((list) => {
+        for (var n = 0; n < list.length; n++)
+          clearEntry(list[n]);
+      });
+      measures.clear();
+    }
+  }
+
+  [kInspect]() {
+    return {
+      timeOrigin,
+      nodeTiming,
+      nodeFrame
+    };
+  }
+}
+
+Object.defineProperties(Performance.prototype, {
+  nodeTiming: { configurable: true, enumerable: true },
+  nodeFrame: { configurable: true, enumerable: true },
+  timeOrigin: { configurable: true, enumerable: true },
+  now: { configurable: true, enumerable: true },
+  getEntries: { configurable: true, enumerable: true },
+  getEntriesByName: { configurable: true, enumerable: true },
+  getEntriesByType: { configurable: true, enumerable: true },
+  mark: { configurable: true, enumerable: true },
+  clearMarks: { configurable: true, enumerable: true },
+  measure: { configurable: true, enumerable: true },
+  clearMeasures: { configurable: true, enumerable: true },
+});
+
+module.exports = {
+  performance: new Performance(),
```

Comment:
do you mean having `module.exports = new Performance()` instead? The challenge with that is the `PerformanceObserver` class, which also hangs off module.exports here.